### PR TITLE
ENC: add AUD NAL unit at the beginning of pic

### LINKED ISSUE #32: Add AUD nal unit at the beginning of the picture for AVC encoder
this will add AUD nal unit

### PR DESCRIPTION
**Pull request by QuPengfei**
Fixes #32

Signed-off-by: Pengfei Qu <Pengfei.Qu@intel.com>

### CODE REVIEW

File: encode/avcenc.c [hunk 1]
```diff
@@ -754,6 +762,32 @@ static int begin_picture(FILE *yuv_fp, int frame_num, int display_num, int slice
...
+                                   sizeof(packed_header_param_buffer), 1, &packed_header_param_buffer,
+                                   &avcenc_context.packed_aud_header_param_buf_id);
+        CHECK_VASTATUS(va_status,"vaCreateBuffer");
+
+        va_status = vaCreateBuffer(va_dpy,
+                                   avcenc_context.context_id,
+                                   VAEncPackedHeaderDataBufferType,
+                                   (length_in_bits + 7) / 8, 1, packed_buffer,
+                                   &avcenc_context.packed_aud_buf_id);
+        CHECK_VASTATUS(va_status,"vaCreateBuffer");
```
Comment (xhaihao) [inline encode/avcenc.c]: Please release the memory pointed by packed_buffer

File: encode/avcenc.c [hunk 2]
```diff
@@ -928,6 +962,12 @@ int avcenc_render_picture()
     if (avcenc_context.misc_parameter_roi_buf_id != VA_INVALID_ID)
         va_buffers[num_va_buffers++] =  avcenc_context.misc_parameter_roi_buf_id;
 
+    if (avcenc_context.packed_aud_header_param_buf_id != VA_INVALID_ID)
+        va_buffers[num_va_buffers++] =  avcenc_context.packed_aud_header_param_buf_id;
+
+    if (avcenc_context.packed_aud_buf_id != VA_INVALID_ID)
+        va_buffers[num_va_buffers++] =  avcenc_context.packed_aud_buf_id;
+
```
Comment (xhaihao) [inline encode/avcenc.c]: Please submit the packed AUD header prior to SPS header

Comment (QuPengfei) [inline encode/avcenc.c]: here the order does not matter. the real submit occurred at the begin_picture().

Anyway, i put them at the beginning of the array va_buffers.

Comment (xhaihao) [inline encode/avcenc.c]: The order is the matter, the buffers are submitted by the following function call. BTW we can't assume the behavior of the driver.

       va_status = vaRenderPicture(va_dpy,
                                avcenc_context.context_id,
                                va_buffers,
                                num_va_buffers);

Comment (QuPengfei) [inline encode/avcenc.c]: yes. right. one good poing is our driver can handle this disorder

**Comment by QuPengfei** [comment]
update the patch

**Pull request merged**